### ISSUE TITLE
No error message if test email fails to send pulse

### ISSUE THREAD

**Issue by draven72** [metabase/metabase#1692]
When setting up a pulse if you click on the send test button, you don't get an error message if the send fails, but an error does appear in the console/log - eg:

[WARN ] metabase.email :: Failed to send email: 530 5.7.57 SMTP; Client was not authenticated to send anonymous mail during MAIL FROM

**Comment by tlrobinson** [comment]
Also applies to sending invite emails.

**Comment by kdoh** [comment]
Just want to check in and make sure we're all on the same page in terms of what we mean by surfacing error messages. The first problem here is that we're not handling an "error" state at all for either pulse test or invites in the case there's a problem sending the email. So that part makes total sense to me.

The second part which I'm a little less clear about is of that is how specific we want to be about what when wrong. It'll likely change from case to case but I wouldn't imagine we'd be able to reliably parse a useful error message and even if we could would we want to display that? I'm fairly against mentioning things like SMTP settings or MAIL FROM since that stuff is scary and no real humans should see messages like that.

So to be concrete, is it enough in the case of pulses to say "Couldn't send email" or was the intention here to try and get to the root of why as part of the messaging? 

For my 💰 I'd say we probably should stick to just letting folks know something went wrong with the test or invite email unless we think we can get terse and human readable enough messages reliably.

**Comment by salsakran** [comment]
I'm coming at this from the other side. A "something went wrong" message doesn't do anyone much good in terms of debugging. We need to be able to get the actual message from people on discourse, etc. 

While a friendly wrapper would be useful, it's *really* important that we don't dance the same dance we did with query builder errors - where we started off with vague friendly errors, and then under the deluge of support problems, had to include the actual error messages.

**Comment by tlrobinson** [comment]
The biggest problem is we silently fail entirely. The email settings "send test email" does show useful error messages, so maybe it would be good enough to let the user know when an email failed and point them in the right direction (user => ask an admin, admin => go to email settings).

**Comment by mazameli** [comment]
Couple ideas/options that depend on if it's important to copy/paste the error text or if a screenshot is sufficient.

**1. Tooltip:**
![tooltip](https://cloud.githubusercontent.com/assets/2223916/16537092/0031df58-3fb0-11e6-95b3-0469d0f5adf4.png)

**2. Click on `Couldn't send` to open modal with full error text**
Better for cases when the error text is super long. Also lets you copy/paste
![modal with error](https://cloud.githubusercontent.com/assets/2223916/16537097/20eb29a2-3fb0-11e6-9091-a4afe7e464d8.png)

**3. Bonus weird idea: a console panel that slides up from bottom of screen when you click `couldn't send`**
![console](https://cloud.githubusercontent.com/assets/2223916/16537117/55424618-3fb0-11e6-8dd2-d9109cff5016.png)

**Comment by salsakran** [comment]
👎  on the tool tip.

We don't know how large or well formatted the errors will be.

**Comment by mazameli** [comment]
So the simplest thing to do would be to go the modal route since we already have that pattern established. Going with the footer console thing might be an idea to explore more broadly at some point, but it would require rethinking our error messages across the app, because we don't want to have one weird spot where the app does something different than everywhere else.

**Comment by salsakran** [comment]
also can we replace "send email error" with "We encountered an error sending your email"?

**Comment by mazameli** [comment]
Yeah, good catch. I wrote, "There was a problem sending this email," which I think is a little friendlier.

![modal with error](https://cloud.githubusercontent.com/assets/2223916/16626922/d940e944-435f-11e6-85ee-556a3a5bf66e.png)

So the click target to bring up this modal should be the `Couldn't send` text along with the question icon. And whomever implements this, please remember `cursor: pointer` on hover. :)

**Comment by brunobergher** [comment]
@KorowkaMu old but relevant
